### PR TITLE
Blockthrough Bid Adapter: initial release

### PR DESCRIPTION
**Pull request by PavloMalashnyak**
## Type of change

- [x] New bidder adapter   

## Description of change

- contact email: engsupport@blockthrough.com
- test parameters for validating bids:
```
{
  bidder: 'blockthrough',
  params: {
    pubmatic: {
       pnlisherId: 'testId'
    }
  }
}
```

## Other information
https://github.com/prebid/prebid.github.io/pull/5050

### CODE REVIEW

File: modules/BTBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
+import { config } from '../src/config.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { deepSetValue, isPlainObject, logWarn } from '../src/utils.js';
+import { BANNER } from '../src/mediaTypes.js';
+import { ortbConverter } from '../libraries/ortbConverter/converter.js';
+
+const BIDDER_CODE = 'blockthrough';
+const GVLID = 815;
+// reach out for approval to marketing@blockthrough.com
+const ENDPOINT_URL = 'https://pbs.sample.com';
```
Comment (patmmccann) [inline modules/BTBidAdapter.js:10]: what is this fake value?

File: modules/BTBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+const SYNC_URL = 'https://cdn.sample.com';
+
+const CONVERTER = ortbConverter({
+  context: {
+    netRevenue: true,
+    ttl: 30,
```
Comment (patmmccann) [inline modules/BTBidAdapter.js:16]: why so low

File: modules/BTBidAdapter.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+```javascript
+pbjs.setBidderConfig({
+  bidders: ['blockthrough'],
+  config: {
+    ortb2: {
+      site: {
+        ext: {
+          blockthrough: {
+            ab: false,
```
Comment (patmmccann) [inline modules/BTBidAdapter.md:33]: can you define these values

File: test/spec/modules/BTBidAdapte_spec.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
+import { expect } from 'chai';
```
Comment (patmmccann) [inline test/spec/modules/BTBidAdapte_spec.js:1]: did you misspell the file name?

File: modules/BTBidAdapter.js [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+function buildRequests(validBidRequests, bidderRequest) {
+  const data = CONVERTER.toORTB({
+    bidRequests: validBidRequests,
+    bidderRequest,
+  });
+
+  return [
+    {
+      method: 'POST',
+      url: ENDPOINT_URL,
```
Comment (patmmccann) [inline modules/BTBidAdapter.js:110]: you need a real url

**Review by patmmccann — CHANGES_REQUESTED** [review]

File: modules/BTBidAdapter.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+# Bid Params
+
+| Key    | Scope    | Type   | Description                                                    |
+| ------ | -------- | ------ | -------------------------------------------------------------- |
+| bidder | Required | Object | Bidder configuration. Could configure several bidders this way |
+
+# Bidder Config
+
+Make sure to set required ab, orgID, websiteID values received after approval using `pbjs.setBidderConfig`.
```
Comment (mkendall07) [inline modules/BTBidAdapter.md:21]: we should include an example of this

File: modules/BTBidAdapter.js (hunk 5 above)
Comment (mkendall07) [reply modules/BTBidAdapter.js:110]: is that a requirement? I think we wanted to gate it behind approval.

File: modules/BTBidAdapter.js (hunk 1 above)
Comment (PavloMalashnyak) [reply modules/BTBidAdapter.js:10]: We want to provide real endpoint values only after publisher approval. Are they required?

File: modules/BTBidAdapter.js (hunk 5 above)
Comment (patmmccann) [reply modules/BTBidAdapter.js:109]: this doesnt appear to fall back to config; can you have a sensible default and fallback?

File: modules/BTBidAdapter.js (hunk 1 above)
Comment (patmmccann) [reply modules/BTBidAdapter.js:9]: am i missing the line where you overwrite this value with the value in configuration?

Comment (mkendall07) [reply modules/BTBidAdapter.js:9]: @patmmccann no we left it out intentionally. I'll talk to the team about putting in the real endpoint.

Comment (patmmccann) [reply modules/BTBidAdapter.js:9]: I must be missing something, is your intention for the publisher to fork the project and change this constant?

Comment (mkendall07) [reply modules/BTBidAdapter.js:9]: @PavloMalashnyak please update the endpoint with the real one. Thanks

**Review by mkendall07 — APPROVED** [review]
LGTM. @patmmccann let me know if you have any concerns before merging

**Pull request merged**